front/app/components/AssignBudgetControl/index.tsx

```diff
@@ -98,184 +87,181 @@ const StyledPBExpenses = styled(PBExpenses)`
-                ),
-                idea_ids: [idea.id],
-              });
-              done();
-              trackEventByName(tracks.basketCreated);
-            } catch (error) {
-              done();
-            }
+          // if signed up & in
+        } else if (!isNilOrError(authUser)) {
+          if (budgetingDisabledReason === 'not_verified') {
+            openVerificationModal({
+              context: {
+                action: 'budgeting',
+                id: participationContextId,
+                type: participationContextType,
+              },
+            });
+          } else if (isBudgetingEnabled) {
+            assignBudget(idea, participationContextId, authUser);
           }
         }
+      };
+
+    const assignBudget = async (
+      idea: IIdeaData,
+      participationContextId: string,
+      authUser: IUserData
+    ) => {
+      const timeout = (ms: number) =>
+        new Promise((resolve) => setTimeout(resolve, ms));
+      const done = async () => {
+        await timeout(200);
+        setProcessing(false);
+      };
+
+      setProcessing(true);
+
+      if (!isNilOrError(basket)) {
+        const basketIdeaIds = basket.relationships.ideas.data.map(
+          (idea) => idea.id
+        );
+        const isInBasket = includes(basketIdeaIds, ideaId);
+        let newIdeas: string[] = [];
+
+        if (isInBasket) {
+          newIdeas = basket.relationships.ideas.data
+            .filter((basketIdea) => basketIdea.id !== idea.id)
+            .map((basketIdea) => basketIdea.id);
+        } else {
+          newIdeas = [
+            ...basket.relationships.ideas.data.map(
+              (basketIdea) => basketIdea.id
+            ),
+            idea.id,
+          ];
+        }
+
+        try {
+          await updateBasket(basket.id, {
+            user_id: authUser.id,
+            participation_context_id: participationContextId,
+            participation_context_type: capitalizeParticipationContextType(
+              participationContextType
+            ),
+            idea_ids: newIdeas,
+            submitted_at: null,
+          });
+          done();
+          trackEventByName(tracks.ideaAddedToBasket);
+        } catch (error) {
+          done();
+          streams.fetchAllWith({ dataId: [basket.id] });
+        }
+      } else {
+        try {
+          await addBasket({
+            user_id: authUser.id,
+            participation_context_id: participationContextId,
+            participation_context_type: capitalizeParticipationContextType(
+              participationContextType
+            ),
+            idea_ids: [idea.id],
+          });
+          done();
+          trackEventByName(tracks.basketCreated);
+        } catch (error) {
+          done();
+        }
       }
     };
 
     if (
-      !isUndefined(authUser) &&
       !isNilOrError(idea) &&
       !isUndefined(basket) &&
-      idea.attributes.budget
+      idea.attributes.budget &&
+      participationContextId
```

Comment:
In the wrapper component, we did this check to ensure participationContextId that was passed was a string. By doing this check here (and passing the string to functions, the other places where it was used), we have the same effect.